impress-org/givewp-next-gen
# Refactor: consolidate app styles and remove shadow dom

## Description



**Shadow DOM removal**

There were some issues with the form builder using a shadow dom and how the app and component styles were being loaded.  At this time we decided the shadow dom is not needed anymore. This lets components bring in there own styles and we don't have to worry about including them separately.

**Consolidate Styles**

We now just have one `App.scss` file that gets loaded into our main `App.tsx` component.  The rest of the component styles are included as needed/used.  

**Basic form preview loading**

To make sure all this was working as expected, a `<Spinner />` component from `@wordpress/components` is now being used to show the loading state of a form preview.  We can revisit this visual loader when there are more settings.

**Fix section height growing when selected**

The inner block appender has been switched to the small button and is positioned in such a way that does not add any height to the section.


## Affects



The form builder styles

## Visuals



**Basic form preview loading:**

![2022-11-10 15 37 44](https://user-images.githubusercontent.com/10138447/201200918-fce7be0c-7027-4ff5-934a-db88835b4a6c.gif)


**No more section height jumping:**

![2022-11-10 15 16 04](https://user-images.githubusercontent.com/10138447/201197034-30e5fe47-a8eb-4498-a5cf-fec99c628ee4.gif)

## Testing Instructions



Build assets and make sure the form builder still looks as expected.

Also test the loading state in the design tab and section appender update.

## Pre-review Checklist
